File: code_eval/cpp/easy/challenge_08.cpp
// Created by cschalk on 3/21/16.
// challenge 08 : Reverse string word order.
#include <iostream>
#include <fstream>
#include <sstream>
#include <stack>

using namespace std;

int main(int argc, char ** argv) {

    std::ifstream input {argv[1]};
    std::string line;

    while (std::getline(input, line)) {
        std::stack<std::string> reversed;
        std::string word;
        std::stringstream ss(line);
        while (ss >> word) {
            reversed.push(word);
            reversed.push(" ");
        }
        reversed.pop(); // lose the first space;
        while(!reversed.empty()) {
            std::cout << reversed.top();
            reversed.pop();
        }
        std::cout << std::endl;
        flush(std::cout);
    }

    return 0;
}

File: code_eval/java/easy/Challenge08.java
package easy;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;

/**
 * Challenge 08 : Reverse Words
 * Created by cschalk on 3/21/16.
 */
public class Challenge08 {

    public static void main (String[] args) throws Exception {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            reverseSentence(line);
        }
    }

    private static void reverseSentence(String sentence) {
        String[] words = sentence.split(" ");
        String reverse = "";
        for (String word : words) {
            reverse = word + " " + reverse;
        }
        System.out.println(reverse.trim());
    }
}
